functions-and-scope.js

```diff
@@ -40,18 +70,68 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
 // Log het antwoord in de terminal.
 
+let sum = 0;
+for (let i = 0; i < grades.length ; i++) {
+    sum = sum + grades[i];
+    }
+const average = sum/grades.length;
+console.log(average);
+
 // ---- Verwachte uitkomst: 6.642857142857143
 
 
 /* 2b: Omschrijven tot een herbruikbare functie */
 // Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
+function averageGrade(arrayCijfers) {
+    let sum = 0;
+    for (let i = 0; i < arrayCijfers.length ; i++) {
+        sum = sum + arrayCijfers[i];
+    }
+    const average = sum/arrayCijfers.length;
+    return  average;
+    //console.log(average);
+}
+
+const gemEen = averageGrade(grades);
+console.log(gemEen);
+
+const gemTwee = averageGrade([6,4,5]);
+console.log(gemTwee);
+
+const gemDrie = averageGrade([8,9,4,6,10]);
+console.log(gemDrie);
 
 // ---- Verwachte uitkomsten:
 // averageGrade(grades) geeft 6.642857142857143
-// averageGrade([6, 4, 5]) geeft xxxx
-// averageGrade([8, 9, 4, 6, 10]) geeft xxxx
+// averageGrade([6, 4, 5]) geeft 5
+// averageGrade([8, 9, 4, 6, 10]) geeft 7.4
+
+const gemEenC = Math.round(gemEen*100)/100;
+console.log(gemEenC);
+///Dit kan echt niet de bedoeling zijn
+const gemTweeC = Math.round(gemTwee*100)/100;
+console.log(gemTweeC);
+console.log(gemTwee.toFixed(2));
+
+//onderstaande geeft wel in twee decimalen, maar is strong geworden
+const gemDrieC = gemDrie.toFixed(2);
+console.log(gemDrieC);
+console.log(typeof gemDrieC);
+console.log(Number(gemDrieC));
+console.log(parseFloat(gemDrieC));
+// ik geef het op ik blijf op een decimaal terugkomen......
+//onderstaande geeft duidelijk het verschil tussen nummer en getal weer
+console.log(gemTweeC+gemDrieC);
+
+/*
+
+console.log(Number(gemDrie.toFixed(2)));
+console.log(Number(gemDrieC));
+console.log(Number(gemTweeC));
+
+*/
```

Comment:
Mijn code: 
function avgGrade(grades) {
    let sumOfGrades = 0;

    for (let i = 0; i < grades.length; i++) {
        sumOfGrades += grades[i];
    }

    const avgGrade = sumOfGrades / grades.length;

    return avgGrade.toFixed(2);
}

console.log(avgGrade(grades));
console.log(avgGrade([6, 4, 5]));
console.log(avgGrade([8, 9, 4, 6, 10]));